### PR TITLE
CPLYTM-418: add the scan subcommand to the complytime CLI

### PR DESCRIPTION
**Pull request by jpower432**
## What Changed?

- Packages that exist just to support the CLI were moved to `internal`.
- A `complytime` package was added to hold core logic
- The scan subcommand was added

In draft until tested with a prototype plugin

### CODE REVIEW

**Comment by jpower432** [comment]
Putting this back in draft because rebasing a branch on a dependency is going to require this to be changed.

**Review by d10n — DISMISSED** [review]

File: internal/complytime/configuration_test.go
Comment (d10n) [inline internal/complytime/configuration_test.go:1]: Just needs a license comment

**Review by d10n — DISMISSED** [review]

**Review by gvauter — DISMISSED** [review]
lgtm

File: cmd/complytime/option/common.go [hunk 1]
```diff
@@ -8,20 +8,21 @@ import (
 	"github.com/spf13/pflag"
 )
 
+// Common options for the ComplyTome CLI.
```
Comment (hbraswelrh) [inline cmd/complytime/option/common.go]: Just change typo to use ComplyTime

File: cmd/complytime/option/common.go [hunk 2]
```diff
@@ -8,20 +8,21 @@ import (
...
-	In io.Reader
+// Output options for
+type Output struct {
 	// Out think, os.Stdout
 	Out io.Writer
 	// ErrOut think, os.Stderr
 	ErrOut io.Writer
 }
 
+// BindFlags populate Common options from a user-specified flags.
```
Comment (hbraswelrh) [inline cmd/complytime/option/common.go]: Nit pick- delete "a" for clarity

**Review by hbraswelrh — CHANGES_REQUESTED** [review]
LGTM

**Review by hbraswelrh — APPROVED** [review]
LGTM

**Review by d10n — APPROVED** [review]

**Review by gvauter — APPROVED** [review]
lgtm

**Pull request merged**